src/dft/backends/portfft/commit.cpp

```diff
@@ -0,0 +1,320 @@
+    std::array<storage_type<descriptor_type>, 2> committed_descriptors = { std::nullopt,
+                                                                           std::nullopt };
+
+public:
+    portfft_commit(sycl::queue& queue, const dft::detail::dft_values<prec, dom>&)
+            : oneapi::mkl::dft::detail::commit_impl<prec, dom>(queue, backend::portfft) {
+        if constexpr (prec == dft::detail::precision::DOUBLE) {
+            if (!queue.get_device().has(sycl::aspect::fp64)) {
+                throw mkl::exception("DFT", "commit", "Device does not support double precision.");
+            }
+        }
+    }
+
+    void commit(const dft::detail::dft_values<prec, dom>& config_values) override {
+        // not available in portFFT:
+        if (config_values.workspace != config_value::ALLOW) {
+            throw mkl::unimplemented("dft/backends/portfft", __FUNCTION__,
+                                     "portFFT only supports ALLOW for the WORKSPACE parameter");
+        }
+        if (config_values.ordering != config_value::ORDERED) {
+            throw mkl::unimplemented("dft/backends/portfft", __FUNCTION__,
+                                     "portFFT only supports ORDERED for the ORDERING parameter");
+        }
+        if (config_values.transpose) {
+            throw mkl::unimplemented("dft/backends/portfft", __FUNCTION__,
+                                     "portFFT does not supported transposed output");
+        }
+
+        // forward descriptor
+        pfft::descriptor<scalar_type, domain> fwd_desc(
+            { config_values.dimensions.cbegin(), config_values.dimensions.cend() });
+        fwd_desc.forward_scale = config_values.fwd_scale;
+        fwd_desc.backward_scale = config_values.bwd_scale;
+        fwd_desc.number_of_transforms =
+            static_cast<std::size_t>(config_values.number_of_transforms);
+        fwd_desc.complex_storage = config_values.complex_storage == config_value::COMPLEX_COMPLEX
+                                       ? pfft::complex_storage::COMPLEX
+                                       : pfft::complex_storage::REAL_REAL;
+        fwd_desc.placement = config_values.placement == config_value::INPLACE
+                                 ? pfft::placement::IN_PLACE
+                                 : pfft::placement::OUT_OF_PLACE;
+        fwd_desc.forward_offset = static_cast<std::size_t>(config_values.input_strides[0]);
+        fwd_desc.backward_offset = static_cast<std::size_t>(config_values.output_strides[0]);
+        fwd_desc.forward_strides = { config_values.input_strides.cbegin() + 1,
+                                     config_values.input_strides.cend() };
+        fwd_desc.backward_strides = { config_values.output_strides.cbegin() + 1,
+                                      config_values.output_strides.cend() };
+        // guard against overwriting the default value when not needed, this seems to lead to issues.
```

Comment:
Can we investigate more what the issue is? portFFT shouldn't care about the distances set when the number of transforms is 1.

Reply:
I guess this has been fixed since I first added the guard https://github.com/FMarno/oneMKL/pull/1/commits/fc720ead06549335a71310bb2dc44e976703488b